lib/ddtrace/ext/transport.rb

```diff
@@ -8,6 +8,7 @@ module HTTP
         ENV_DEFAULT_PORT = 'DD_TRACE_AGENT_PORT'.freeze
         ENV_DEFAULT_URL = 'DD_TRACE_AGENT_URL'.freeze
         HEADER_CONTAINER_ID = 'Datadog-Container-ID'.freeze
+        HEADER_DD_API_KEY = 'DD-API-KEY'.freeze
```

Comment:
Would this key be a generic key to be used in possible future products? (It's ok if we don't know the answer yet)